addons/l10n_de/models/res_company.py

```diff
@@ -20,7 +20,7 @@ class ResCompany(models.Model):
     def write(self, vals):
         if (
             'account_fiscal_country_id' in vals
-            and (german_companies := self.filtered(lambda c: c.account_fiscal_country_id.code == 'DE'))
+            and (german_companies := self.filtered(lambda c: c.account_fiscal_country_id.code == 'DE' and c.account_fiscal_country_id.id != vals.get('account_fiscal_country_id')))
             and self.env['account.move'].search_count([('company_id', 'in', german_companies.ids)], limit=1)
         ):
             raise ValidationError(_("You cannot change the fiscal country."))
```

Comment:
With this additional condition inside the if clause, I think it could be clearer if we refactor the method a bit:

```suggestion
    def write(self, vals):
        if 'account_fiscal_country_id' in vals:
            german_companies = self.filtered(
                lambda c: c.account_fiscal_country_id.code == 'DE'
                and c.account_fiscal_country_id.id != vals['account_fiscal_country_id']
            )
            if german_companies and self.env['account.move'].search_count([('company_id', 'in', german_companies.ids)], limit=1):
                raise ValidationError(_("You cannot change the fiscal country."))
        return super().write(vals)
```

If you prefer not to, I'd suggest breaking this line down, it's a bit too long.
```
and (german_companies := self.filtered(
        lambda c: c.account_fiscal_country_id.code == 'DE'
        and c.account_fiscal_country_id.id != vals['account_fiscal_country_id']
    )
)
```

Reply:
@etobella My bad with this suggestion :/ The comment by the final reviewer has the best approach with minimal diff but still being quite readable.